cogeo_mosaic/backends/base.py

```diff
@@ -136,19 +142,59 @@ def update(
 
     def assets_for_tile(self, x: int, y: int, z: int) -> List[str]:
         """Retrieve assets for tile."""
-        return self.get_assets(x, y, z)
+        mosaic_tms = self.mosaic_def.tilematrixset or WEB_MERCATOR_TMS
+        if self.tms.rasterio_crs == mosaic_tms.rasterio_crs:
+            return self.get_assets(x, y, z)
+
+        xmin, ymin, xmax, ymax = self.tms.bounds(x, y, z)
```

Comment:
use geographic coordinates or TMS's crs ?

Reply:
was a bit confused by this from prior prs, if tms.bounds returns coordinates in the crs of the tms then it should be fine as is. Using one or the other shouldn't matter so long as they are used with consistency